extensions/steve/steve.go

```diff
@@ -36,3 +38,45 @@ func WaitForSteveResourceDeletion(client *rancher.Client, interval, timeout time
 
 	return nil
 }
+
+// WaitForSteveResourceCreation waits for a given steve object to be created/come up active.
+func WaitForSteveResourceCreation(client *rancher.Client, v1Resource *v1.SteveAPIObject, interval, timeout time.Duration) error {
+	err := kwait.PollUntilContextTimeout(context.TODO(), interval, timeout, true, func(ctx context.Context) (done bool, err error) {
+		client, err = client.ReLogin()
+		if err != nil {
+			return false, err
+		}
+
+		clusterResp, err := client.Steve.SteveType(v1Resource.Type).ByID(v1Resource.ID)
+		if err != nil {
+			return false, err
+		}
+
+		if clusterResp.ObjectMeta.State.Name == active {
+			logrus.Infof("%s(%s) is active", v1Resource.Kind, v1Resource.Name)
+			return true, nil
+		}
+
+		return false, nil
+	})
+
+	return err
+}
+
+// CreateResource creates a steve resource and polls the resulting object until it comes up active.
+func CreateResource(client *rancher.Client, v1ResourceType string, v1Resource any, poll bool, interval, timeout time.Duration) (*v1.SteveAPIObject, error) {
```

Comment:
Can we name this `CreateAndWaitForResource`? Sort of to distinguish why some one would choose this extension over the client create. In the near future we will have generated steve clients as well, just an FYI

Reply:
I am fine with this, I'll update it.